packages/zero-runtime/src/processors/css.ts

```diff
@@ -0,0 +1,179 @@
+      this.handleTemplate(this.callParam, values);
+    } else {
+      this.handleCall(this.callParam, values);
+    }
+  }
+
+  private handleTemplate([, callArgs]: TemplateParam, values: ValueCache) {
+    const templateStrs: string[] = [];
+    const templateExpressions: Primitive[] = [];
+    callArgs.forEach((item) => {
+      if ('kind' in item) {
+        switch (item.kind) {
+          case ValueType.FUNCTION:
+            throw item.buildCodeFrameError(
+              'MUI: Functions are not allowed to be interpolated in css tag.',
+            );
+          case ValueType.CONST:
+            templateExpressions.push(item.value);
+            break;
+          case ValueType.LAZY: {
+            const evaluatedValue = values.get(item.ex.name);
+            if (typeof evaluatedValue === 'function') {
+              throw item.buildCodeFrameError(
+                'MUI: Functions are not allowed to be interpolated in css tag.',
+              );
+            } else {
+              templateExpressions.push(evaluatedValue as Primitive);
+            }
+            break;
+          }
+          default:
+            break;
+        }
+      } else if (item.type === 'TemplateElement') {
+        templateStrs.push(item.value.cooked as string);
+      }
+    });
+    this.generateArtifacts(templateStrs, ...templateExpressions);
+  }
+
+  generateArtifacts(styleObjOrTaggged: CSSInterpolation | string[], ...args: Primitive[]) {
+    const cssClassName = css(styleObjOrTaggged, ...args);
+    const cssText = cache.registered[cssClassName] as string;
+
+    const rules: Rules = {
+      [this.asSelector]: {
+        className: this.className,
+        cssText,
+        displayName: this.displayName,
+        start: this.location?.start ?? null,
+      },
+    };
+    const sourceMapReplacements: Replacements = [
+      {
+        length: cssText.length,
+        original: {
+          start: {
+            column: this.location?.start.column ?? 0,
+            line: this.location?.start.line ?? 0,
+          },
+          end: {
+            column: this.location?.end.column ?? 0,
+            line: this.location?.end.line ?? 0,
+          },
+        },
+      },
+    ];
+    this.artifacts.push(['css', [rules, sourceMapReplacements]]);
+  }
+
+  private handleCall([, ...callArgs]: CallParam, values: ValueCache) {
+    const mergedStyleObj: CSSInterpolation = {};
+
+    callArgs.forEach((callArg) => {
+      let styleObj: CSSInterpolation;
+      if (callArg.kind === ValueType.LAZY) {
+        styleObj = values.get(callArg.ex.name) as CSSInterpolation;
+      } else if (callArg.kind === ValueType.FUNCTION) {
+        const { themeArgs } = this.options as IOptions;
+        const value = values.get(callArg.ex.name) as (
+          args: Record<string, unknown> | undefined,
+        ) => CSSInterpolation;
+        styleObj = value(themeArgs);
+      }
+
+      if (styleObj) {
+        deepMerge(mergedStyleObj, styleObj);
```

Comment:
What happens if the return is literal template?

```js
css(({ theme }) => `color: ${theme.vars.palette.divider;`;                                                         
```

Reply:
Yeah. This won't work. This is the reason I kept it simple for now. Looks like I'll have to cover multiple cases.

Reply:
Updated the implementation.